protobuf/api/execution.proto

```diff
@@ -70,6 +70,11 @@ message CreateExecutionRequest {
     (gogoproto.customtype) = "github.com/mesg-foundation/engine/hash.Hash",
     (gogoproto.nullable) = false
   ];
+
+  // price of running the execution.
+  string price = 10 [
```

Comment:
Can't we use the type `Coin` from cosmos here instead of having to parse it, this will be much easier to interact with if we have the same struct for all the messages/resources

Reply:
I thought about that, but I haven't seen Unmarshal for type `Coins`. I can test it today.

Reply:
I'm waiting for the status PR status https://github.com/cosmos/cosmos-sdk/issues/5685. Without it we have drop support for equal_all in executions.proto.

Reply:
Looks like the support of `equal_all` has been added 🎉 https://github.com/cosmos/cosmos-sdk/pull/5688
For coin but also all the other cosmos types.

Reply:
Can we at least use their proto types instead of a string?